### ISSUE TITLE
New samples for fsharp based on version 2

### ISSUE THREAD

**Issue by darting** [dotnet/orleans#3845]
Hi there,

can we have a new sample for fsharp which based on version 2? thanks :)

**Comment by darting** [comment]
I was usd  `parts.AddApplicationPart((typeof<HelloGrain>).Assembly).WithCodeGeneration()` to configure host but got `Orleans.Runtime.OrleansException: Cannot find an invoker for interface IHelloGrain`. and after I changed to  `parts.AddFromAppDomain().WithCodeGeneration()` it is working now.

**Comment by ReubenBond** [comment]
Hi @darting, your repo looks like a good sample, would you be able to contribute it into a folder such as Samples/FSharp.NetCore?

Perhaps using [Giraffe.Tasks](https://github.com/giraffe-fsharp/Giraffe.Tasks) instead of the custom TaskBuilder

**Comment by darting** [comment]
hi @ReubenBond , I had create a PRs for this issue, thanks. :)

**Comment by ReubenBond** [comment]
@darting, you can do this instead:
```fsharp
parts.AddApplicationPart(typeof<HelloGrain>.Assembly).WithReferences().WithCodeGeneration()
```
or use this better alternative (better since codegen runs only on your assemblies and not on their references):
```fsharp
parts.AddApplicationPart(typeof<HelloGrain>.Assembly).WithCodeGeneration()
parts.AddApplicationPart(typeof<IHelloGrain>.Assembly).WithCodeGeneration()
```

**Comment by darting** [comment]
hey @ReubenBond thanks for your suggestion. already updated.

**Comment by ReubenBond** [comment]
Fixed via #3846, thanks @darting! :)
